### ISSUE TITLE
make -j <N> seem to be broken

### ISSUE THREAD

**Issue by ashinkarov** [coccinelle/coccinelle#95]
It seems that parallel make is broken.  I got the following errors when compiling with "make -j7":

```
File "cocci_vs_c.mli", line 1:
Error: Corrupted compiled interface
../parsing_c/control_flow_c.cmi
...
File "popl.mli", line 1:
Error: Corrupted compiled interface
../ctl/wrapper_ctl.cmi
```
I used a snapshot of the master (2017-04-22).  It also fails when building a package on Gentoo, where "-j<n>" usually is a default option.  Sequential make works fine though.

Could it be that some dependencies are missing between modules?  I am not very familiar with the codebase, so it is not immediately obvious to me.

**Comment by JuliaLawall** [comment]
On Sat, 22 Apr 2017, Artem Shinkarov wrote:

>
> It seems that parallel make is broken. I got the following errors when
> compiling with "make -j7":
>
> File "cocci_vs_c.mli", line 1:
> Error: Corrupted compiled interface
> ../parsing_c/control_flow_c.cmi
> ...
> File "popl.mli", line 1:
> Error: Corrupted compiled interface
> ../ctl/wrapper_ctl.cmi
>
> I used a snapshot of the master (2017-04-22). It also fails when building a
> package on Gentoo, where "-j" usually is a default option. Sequential make
> works fine though.
>
> Could it be that some dependencies are missing between modules? I am not
> very familiar with the codebase, so it is not immediately obvious to me.

Could you try make distclean?  The following worked fine for me:

   102	13:41	git pull
   103	13:41	make distclean
   104	13:41	make
   105	13:41	./autogen
   106	13:41	./configure
   107	13:41	make -j8

julia

**Comment by bebehei** [comment]
> Could you try make distclean? The following worked fine for me:

Sorry, but I don't think a `make distclean` should be important. It isn't working either.

I'd claim these steps should be sufficient to reproduce it:

```
git clone https://github.com/coccinelle/coccinelle coccinelle-build
cd coccinelle-build
./autogen
./configure
make -j8
```

I'm running on an ArchLinux system with with recent packages and have got a i5-4570 (quadcore) processor. You can interchange the `-j8` with any number except `1`. `make -j1` works.

**Comment by jacob-keller** [comment]
I can reproduce this on the current master branch, using either set of steps, parallel make will fail.

**Comment by jacob-keller** [comment]
File "parsing_cocci/cocci_args.ml", line 119, characters 17-38:
119 |       let line = Stdcompat.String.trim line in
                       ^^^^^^^^^^^^^^^^^^^^^
Error: Unbound module StdcompatOCAMLC    parsing_c/token_c.mli

make: *** [Makefile:422: parsing_cocci/cocci_args.cmx] Error 2
make: *** Waiting for unfinished jobs....

**Comment by JuliaLawall** [comment]
Thanks for the information.  I have asked the person who takes care of the make file to look into it.

**Comment by jacob-keller** [comment]
I think I just figured it out, actually. We have to add parsing_cocci/cocci_args to the STDCOMPAT_USERS variable. I can submit a pull request with the fix.

**Comment by jacob-keller** [comment]
https://github.com/coccinelle/coccinelle/pull/197 fixes the parallel build on my system.

**Comment by JuliaLawall** [comment]
On Tue, 5 May 2020, Jacob Keller wrote:

>
> I think I just figured it out, actually. We have to add
> parsing_cocci/cocci_args to the STDCOMPAT_USERS variable. I can submit a
> pull request with the fix.

Could you send a patch to the mailing list: cocci@systeme.lip6.fr

It should have a Signed-off-by:

thanks,
julia

**Comment by jacob-keller** [comment]
Yea, I will do so tomorrow morning, since I shut down my development system for the day already.

**Comment by JuliaLawall** [comment]
On Tue, 5 May 2020, Jacob Keller wrote:

>
> Yea, I will do so tomorrow morning, since I shut down my development system
> for the day already.

No problem, I applied the patch.

julia

**Comment by vgambier** [comment]
Fixed in  #197
